tmt/result.py

```diff
@@ -204,7 +206,14 @@ def to_serialized(self) -> Dict[str, Any]:
 
     @classmethod
     def from_serialized(cls, serialized: Dict[str, Any]) -> 'Result':
-        return cls(serialized, name=serialized.pop('name'))
+        # Our special key may or may not be present, depending on who
+        # calls this method.  In any case, it is not needed, because we
+        # already know what class to restore: this one.
+        serialized.pop('__class__', None)
+
+        name = serialized.pop('name')
+        serialized['result'] = ResultOutcome.from_spec(serialized['result'])
```

Comment:
"Serialized" is something tmt creates, therefore reading it back should be absolutely safe, no need for `from_spec()` here.

Reply:
Never mind, I see why it's here - we use `from_serialized()` from `check_custom_result()`. That's fine then, I would just add a comment `TODO` comment: `from_serialized()` should trust the input, I suppose we should add its clone to read serialized-but-from-possibly-untrustworthy-source data - that's the place where the schema validation would happen in the future, as we discussed briefly on IRC. So let's add a `TODO` comment, and fix it later, with added protection.